index.iced

```diff
@@ -128,11 +146,59 @@ class Runner
 
   ##-----------------------------------------
 
+  run_test_case_guarded : (code, case_obj, gcb) ->
+    remove_uncaught = () ->
+    timeoutObj = null
+    cb_called = false
+    cb = () =>
+      if timeoutObj
+        clearTimeout(timeoutObj)
+        timeoutObj = null
+
+      unless cb_called
+        cb_called = true
+        remove_uncaught()
+        gcb.apply @, arguments
+
+    format_stack = (err) => format_stack_frame_str @_cur_file_path, err
+
+    if @uncaughtException
+      # "On Error Resume Next"
+      # This is needed, because otherwise testing is stopped after first
+      # failed async test. We want to continue as far as we can, even if in
+      # unstable state (test run is considered "failed" from now on anyway).
+      remove_uncaught = () -> process.removeAllListeners 'uncaughtException'
+      remove_uncaught()
+      process.on 'uncaughtException', (err) ->
+        console.log ":: Recovering from async exception: #{err}"
+        console.log ":: Testing may become unstable from now on."
+        try console.log format_stack(err)
```

Comment:
why is this in a `try`?

Reply:
Because `format_stack` has to access `.stack` property of Error object, and that property is overridable (and iced compiler overrides it to format stack traces to have proper line numbers), I wouldn't want it to throw here and go into some kind of throw-handle infinite loop.

Reply:
OK looks like it was not necessary anyway, because the handler is not called for exceptions inside of it.
```
process.on 'uncaughtException', (err) ->
    throw new Error "aaa"
```
does not go into an infinite loop.